File: algorithms/leetcode/252-meeting-rooms/MeetingRooms.java
class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[1] - b[1]);
        
        int overlap = 0;
        int maxEnd = Integer.MIN_VALUE;
        for (int i = 0; i < intervals.length; i++) {
            if (intervals[i][0] >= maxEnd) {
                maxEnd = intervals[i][1];
            } else {
                overlap++;
            }
        }
        return overlap == 0;
    }
}

File: leetcode/252-meeting-rooms/meeting-rooms.js
const canAttendMeetings = intervals => {
    intervals.sort((a, b) => ((a.start - b.start) || (a.end - b.end)));

    for (let i = 0; i < intervals.length; i++) {
        let end = intervals[i].end;

        if (i < (intervals.length - 1) && (end > intervals[i + 1].start)) {
            return false;
        }
    }

    return true;
};
